src/diffusion/model.py

```diff
@@ -274,3 +279,13 @@ def rsqrt_with_warmup(step: int, max_lr: float, min_lr: float, warmup: int) -> f
     if min_lr is not None:
         lr = max(lr, min_lr)
     return lr
+
+
+def linear(step: int, max_lr: float, min_lr: float, decay_range: tuple) -> float:
+    s1, s2 = decay_range
```

Comment:
The decay range should be int, the number of total steps of decaying. Use warmup to specify the number of steps before starting decay.